### DISCUSSION TITLE
New plugin available: Automatic IPN generator

### DISCUSSION THREAD

**Discussion by LavissaWoW** [inventree/InvenTree#6811] [Show and tell]
Hey folks,

I just released the first version of a plugin that autogenerates IPNs for parts.

https://github.com/LavissaWoW/inventree-ipn-generator

The plugin uses a "regex-like" pattern syntax to define how to generate an IPN.
While "regex-like", the pattern syntax is intuitive and easy to understand.

Patterns consist of any arbitrary combination of the following three categories:
1. Literals - These groups never change, and are useful for prefixes, suffixes, separator characters etc.
`(AB)` - will always be `AB`
3. Characters - Groups of either single chars or ranges of chars.
`[abf]` - Will cycle `a`, `b`, `f`, `a`, `b`, `f` forever
`[a-d]` - Will alphabetically cycle the chars `a` through `d`.
4. Numerics - Groups that define number ranges 
`{3}` - Will loop `001` - `999`
`{3+}` - Will loop `3` - `9` forever

More in-depth examples are available in the plugin's repository

The plugin is available on PyPi, so it can be installed as `inventree-ipn-generator` either via `pip` or your plugins.txt file

**Comment by Century1026** [comment]
Hi Lavissa,

I tested your plug-in, it isn't working. I recorded a video for it. Can you help me figure it out?

https://github.com/inventree/InvenTree/assets/106154819/4a46e7fb-3e0c-48c0-bf18-362f2724bb43



Best,
Century

**Reply by LavissaWoW** [reply]
Check that you have installed version 0.1.1. There was an issue with dashes in version 0.1, resulting in no IPN being generates for patterns containing any kind of dash.

**Comment by eeintech** [comment]
Nice work ! 🥳
